### ISSUE TITLE
Accessibility issues in all versions of GOGS

### ISSUE THREAD

**Issue by ogomez92** [gogits/gogs#3340]
- Gogs version (or commit ref): 0.9.58.0726
- Git version: 2.9.2
- Operating system: Debian 8.1
- Database (use `[x]`):
  - [ ] PostgreSQL
x  - [ ] MySQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io: yes but no, inaccessible captcha
  - [ ] Yes (provide example URL)
  - [ ] No
x  - [ ] Not relevant
- Log gist: irrelevant

## Description

There are multiple accessibility bugs. First, all the mouse over menus. They should be properly made accessible to screen readers using aria, or at least have a setting to turn them off. I can't figure out how to access most pages in GOGS because the screen reader doesn't know that there is a menu there because screen readers don't use mouse cursor to pass over items.

Second, the captcha service you use is not accessible (why not use recaptcha or have a new to define your own captcha via php so I can add a question captcha(?

Thanks.
...

**Comment by bill-auger** [comment]
notabug currently disables captchas for this same reason and has also considered using a more-accessible captcha library - upstream now has ENABLE_CAPTCHA config option but this is an "all or nothing" option

if an alternative captcha library were user then semantically 'ENABLE_CAPTCHA' should probably be true - perhaps an additional config key like 'CAPTCHA_TYPE' could be added to accommodate alternative captcha libs

**Comment by Flameborn** [comment]
These accessibility issues are caused by SemanticUI. Numerous attempts were made to raise awareness (https://github.com/Semantic-Org/Semantic-UI/issues/348, https://github.com/Semantic-Org/Semantic-UI/issues/3652, https://github.com/Semantic-Org/Semantic-UI/issues/3127), unfortunately the developers are not interested in making their product accessible, despite its popularity.

To fix this, you would either have to switch to a different, accessible UI, such as JQWidgets (http://www.jqwidgets.com/), or provide ARIA markup for the various elements. There is ample documentation about ARIA located here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA

Currently, the biggest issue is the dropdown navigation menu, as it would let screen reader users create repositories, select various items on the installation screen, etc. Other elements include: Flashing alerts and dialogs (such as the license selection, when creating a repository), as far as I was able to determine running through the pages.

Being a screen reader user myself, Unfortunately, I will not be able to help, as the end result would be far from pleasing visually. As a temporary solution, style sheets can be disabled in all major browsers. It makes navigation clunky, especially having to navigate through dozens of languages at the bottom, but it works.

GoGS fills in a huge gap, as there is currently no other low-resource Git solution for computers like the Raspberry Pi.

**Comment by lilmike** [comment]
Totally +1. I love gogs, but these issues, especially now that I've moved to linux where (arguably) the screen reader isn't as forgiving on things like this, is almost unusable for me. I can't add a license to a repository or choose a .gitignore because of the inaccessibility of the UI toolkit. If only I knew go then I would help as much as i could, but for now all I think I can offer is testing if needed.
Anyway, if this could be worked around some way this would be great!
-Michael.

**Comment by clarcharr** [comment]
These issues are pretty important and I'd like to see some work on this!

**Comment by IssuehuntBot** [comment]
@issuehuntfest has funded $20.00 to this issue. [See it on IssueHunt](https://issuehunt.io/repos/16752620/issues/3340)

**Comment by dingpengyu** [comment]
关于gogs的无障碍问题，请问@Unknwon 有什么建议没，现在不管是gitea，还是gogs在使用屏幕阅读器的时候体验性真心不好，上面已经有朋友引用了我在gitea发的问题，希望有大佬可以把无障碍这个老大难问题给解决一下谢谢。
最后发一下我的测试环境
操作系统:
windows10-1903
gitea版本
1.8.1
gogs版本:
0.11.86
屏幕阅读器
NVDA alpha-17526
download link:
https://ci.appveyor.com/api/buildjobs/qpdx77b7yh2x64nf/artifacts/output%2Fnvda_snapshot_alpha-17526%2C3ef8410e.exe

**Comment by dingpengyu** [comment]
@nvaccess/nvda: NVDA, the free and open source Screen Reader for Microsoft Windows
https://github.com/nvaccess/nvda

**Comment by ogomez92** [comment]
Any news on this? Is the accessiblity any better nowadays?

**Comment by clarfon** [comment]
There's active discussion on https://github.com/go-gitea/gitea/issues/7057 in case you want to follow that

**Comment by unknwon** [comment]
Chrome's DevTools to explore the accessibility tree: https://developers.google.com/web/tools/chrome-devtools/accessibility/reference#pane

**Comment by landeqiming666** [comment]
I’m working on a focused part of this issue: making the repository creation dropdowns (owner, .gitignore, license, and README) accessible by keyboard and to screen readers. This complements #8259, which targets the signed-in top navigation menus.

The change keeps the existing Semantic UI behavior, adds combobox/listbox semantics and state synchronization, and includes a focused template regression test. I’ll open the pull request shortly with the full validation details.

**Comment by dresecuri-lee** [comment]
Hello Gogs maintainers. Issue #3340 remains open and labeled help wanted, while PRs #8259 and #8389 cover the signed-in navigation and repository-creation dropdowns. Is there an accessibility subtask still uncovered, such as CAPTCHA or another screen-reader flow, and is the IssueHunt-funded $20 reward still active for it? Could you confirm the expected acceptance criteria and payment conditions before work begins? This would be AI-assisted development and review, submitted with this account owner's authorization. I have not claimed the issue.
